File: memoization.fact.py
import time

table = {}

def fact(n):
    if n == 0 or n == 1:
        return 1
    else:
        if n not in table:
            table[n] = n*fact(n-1)
        return table[n]
    
n = int(input("Enter the value: "))
t1 = time.time()
ans = fact(n)
t2 = time.time()
print(f"The {n}th term is {ans}")
print(f"The time taken was {t2-t1} seconds")

File: memoization.fact.c
#include<stdio.h>
long long int table[500];
int fact(int n)
{
    if(n==0||n==1)
    return 1;
    else
    {
        if(table[n]==-1)
        table[n]=n*fact(n-1);
        return table[n];
    }

}
int main()
{
    int n , i;
    printf("Enter the value: ");
    scanf("%d" , &n);
    for(i=0;i<500;i++)
    table[i]=-1;
    printf("the factorial is %d" , fact(n));
    return 0;
}